tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1060 @@
+      "\u001b[0;31mRuntimeError\u001b[0m: batched == nullptr INTERNAL ASSERT FAILED at \"/private/var/folders/fn/c72nxv0x0xj4bzdgq3b0r86r0000gn/T/pip-req-build-ulv3uwfj/functorch/csrc/Interpreter.cpp\":95, please report a bug to PyTorch. "
+     ]
+    }
+   ],
+   "source": [
+    "num_models = 10\n",
+    "\n",
+    "tensordict = TensorDict({\"a\": torch.randn(5, 3)}, batch_size=[5])\n",
+    "\n",
+    "splitlinear_models = [\n",
+    "    TensorDictModule(\n",
+    "        nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"output\"]\n",
+    "    )\n",
+    "    for _ in range(num_models)\n",
+    "]\n",
+    "\n",
+    "\n",
+    "def transpose_stack(tuple_of_tuple_of_tensors):\n",
+    "    tuple_of_tuple_of_tensors = tuple(zip(*tuple_of_tuple_of_tensors))\n",
+    "    results = tuple(\n",
+    "        torch.stack(shards) for shards in tuple_of_tuple_of_tensors\n",
+    "    )\n",
+    "    return results\n",
+    "\n",
+    "\n",
+    "### Let's extract the functional version of the common model\n",
+    "func = splitlinear_models[0].make_functional_with_buffers()[0]\n",
+    "\n",
+    "### We also extract parameters and buffers for every block and stack them together\n",
+    "params, buffers = zip(\n",
+    "    *[\n",
+    "        splitlinear.make_functional_with_buffers()[1]\n",
+    "        for splitlinear in splitlinear_models\n",
+    "    ]\n",
+    ")\n",
+    "params = transpose_stack(params)\n",
+    "buffers = transpose_stack(buffers)\n",
+    "\n",
+    "### Finally we can apply the function to the stack of params and buffer using vmap\n",
+    "func(tensordict, params=params, buffers=buffers, vmap=True).shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "31be6c45-10fb-4fd1-a52f-92214b76c00a",
+   "metadata": {},
+   "source": [
+    "## Do's and don't with `TensorDictModule`\n",
+    "\n",
+    "When `TensorDictModule`, we need to use `TensorDictSequence` if we want to merge it with other operations.\n",
+    "\n",
+    "Do not use `nn.Module` wrappers. This would break some of `TensorDictModule` features such as `functorch` compatibility."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22e65356-d8b3-4197-84b8-598330c1ddc8",
+   "metadata": {},
+   "source": [
+    "## TensorDictModule for RL"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d49a911-933c-476f-8c9a-00e006ed043c",
+   "metadata": {},
+   "source": [
+    "In the context of RL torchrl offers a few wrappers on `TensorDictModule`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e33904a6-d405-45db-a713-47493ca8ee33",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### `ProbabilisticTensorDictModule`"
```

Comment:
To add here:

1. When you print the tensordict, put a few words in the print to say what it is (otherwise the print is not very readable)
2. Add something about interaction mode: how do you switch from "random" to "mode"? What is the difference? What sampling mode exist?
3. Add something about `get_dist`
4. How do you return the parameters of the dist in the tensordict? And the log_prob? Give examples for each
5. What distributions are available in torchrl?

This is ultra important for RL people! You must spend time on these classes

Reply:
Is this not getting a bit far from the tensordictmodule initial tutorial? Won't this need it's own tutorial on its own to dig deeper on RL modules?